src/main/java/at/hannibal2/skyhanni/utils/NumberUtil.kt

```diff
@@ -39,19 +39,57 @@ object NumberUtil {
      * @link https://stackoverflow.com/a/30661479
      * @author assylias
      */
+
     @JvmStatic
-    fun format(value: Number): String {
+    fun format(value: Number, preciseBillions: Boolean = false): String {
         @Suppress("NAME_SHADOWING")
         val value = value.toLong()
-        //Long.MIN_VALUE == -Long.MIN_VALUE, so we need an adjustment here
-        if (value == Long.MIN_VALUE) return format(Long.MIN_VALUE + 1)
-        if (value < 0) return "-" + format(-value)
+        //Long.MIN_VALUE == -Long.MIN_VALUE so we need an adjustment here
+        if (value == Long.MIN_VALUE) return format(Long.MIN_VALUE + 1, preciseBillions)
+        if (value < 0) return "-" + format(-value, preciseBillions)
+
         if (value < 1000) return value.toString() //deal with small numbers
+
         val (divideBy, suffix) = suffixes.floorEntry(value)
+
         val truncated = value / (divideBy / 10) //the number part of the output times 10
-        val truncatedAt = if (suffix == "M") 1000 else 100
+
+        val truncatedAt = if (suffix == "M") 1000 else if (suffix == "B") 1000000 else 100
+
+        val hasDecimal = truncated < truncatedAt && truncated / 10.0 != (truncated / 10).toDouble()
+
+        return if (value > 1_000_000_000 && hasDecimal && preciseBillions) {
+            val decimalPart = (value % 1_000_000_000) / 1_000_000
+            "${truncated / 10}.$decimalPart$suffix"
+        } else {
+            if (hasDecimal) (truncated / 10.0).toString() + suffix else (truncated / 10).toString() + suffix
+        }
+    }
+
+    @JvmStatic
+    fun format3(value: Number, digit: Int): String {
```

Comment:
What is this new method for? Either give it a better method name or add a javadoc-like description on top. Furthermore, would it be possible to merge those 2 methods by any chance? They look pretty similar already.